File: backend/models/DroneService.js
const mongoose = require('mongoose');

const droneServiceSchema = new mongoose.Schema({
  providerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  serviceName: {
    type: String,
    required: true
  },
  description: {
    type: String
  },
  pricePerHa: {
    type: Number,
    required: true
  },
  availableAreas: [
    {
      type: String
    }
  ],
  images: [
    {
      type: String
    }
  ]
}, { timestamps: true });

module.exports = mongoose.model('DroneService', droneServiceSchema);

File: lib/models/drone_service.dart
class DroneService {
  final String id;
  final String providerId;
  final String serviceName;
  final String description;
  final double pricePerHa;
  final List<String> availableAreas;
  final List<String> images;

  DroneService({
    required this.id,
    required this.providerId,
    required this.serviceName,
    required this.description,
    required this.pricePerHa,
    required this.availableAreas,
    required this.images,
  });

  factory DroneService.fromJson(Map<String, dynamic> json) {
    return DroneService(
      id: json['_id'],
      providerId: json['providerId'],
      serviceName: json['serviceName'],
      description: json['description'],
      pricePerHa: json['pricePerHa'].toDouble(),
      availableAreas: List<String>.from(json['availableAreas']),
      images: List<String>.from(json['images']),
    );
  }
}
